src/distances.jl

```diff
@@ -24,53 +32,76 @@ julia> Ellipsoidal([1.0,0.5], [π/2])
 julia> Ellipsoidal([1.0,0.5,0.5], [π/2,0.0,0.0])
 ```
 """
+
+# Right-hand rule; motion defined looking to the negative direction of the axis
+struct RotationRules
+  rot_seq::Symbol
+  motion::Vector{Symbol}
+  radian::Bool
+  main::Symbol
+  extrinsic::Bool
+end
+
+themes = Dict(
+  :TaitBryanExtr => RotationRules(:ZXY,[:CCW,:CCW,:CCW],true,:x,true),
+  :TaitBryanIntr => RotationRules(:ZXY,[:CCW,:CCW,:CCW],true,:x,false),
+  :EulerExtr => RotationRules(:ZXZ,[:CCW,:CCW,:CCW],true,:x,true),
+  :EulerIntr => RotationRules(:ZXZ,[:CCW,:CCW,:CCW],true,:x,false),
+  :GSLIB => RotationRules(:ZXY,[:CW,:CCW,:CCW],false,:y,false),
+  :Leapfrog => RotationRules(:ZYZ,[:CW,:CW,:CW],false,:x,false),
+  :Datamine => RotationRules(:ZXZ,[:CW,:CW,:CW],false,:x,false)
+)
+
 struct Ellipsoidal{N,T} <: Metric
   dist::Mahalanobis{T}
 
-  function Ellipsoidal{N,T}(semiaxes, angles) where {N,T}
+  function Ellipsoidal{N,T}(semiaxes, angles; convention=:TaitBryanExtr) where {N,T}
     @assert length(semiaxes) == N "number of semiaxes must match spatial dimension"
     @assert all(semiaxes .> zero(T)) "semiaxes must be positive"
     @assert N ∈ [2,3] "dimension must be either 2 or 3"
 
+    # invert x and y if necessary
+    if themes[convention].main == :y
+       semiaxes[1], semiaxes[2] = semiaxes[2], semiaxes[1]
+    end
+
     # scaling matrix
     Λ = Diagonal(one(T)./semiaxes.^2)
 
+    # convert to radian if necessary
+    angles = themes[convention].radian ? angles : deg2rad.(angles)
+
     # rotation matrix
     if N == 2
-      θ = angles[1]
-
-      cosθ = cos(θ)
-      sinθ = sin(θ)
-
-      P = [cosθ -sinθ
-           sinθ  cosθ]
+      θ = themes[convention].motion[1] == :CCW ? angles[1] : -1*angles[1]
+      P = angle_to_dcm(θ, 0, 0, themes[convention].rot_seq)[1:2,1:2]
+
+    else # N == 3
+      for (i,sign) in enumerate(themes[convention].motion)
+        angles[i] = sign == :CCW ? angles[i] : -1*angles[i]
+      end
+      P = angle_to_dcm(angles[1], angles[2], angles[3], themes[convention].rot_seq)
     end
-    if N == 3
-      θxy, θyz, θzx = angles
-
-      cosxy = cos(θxy)
-      sinxy = sin(θxy)
-      cosyz = cos(θyz)
-      sinyz = sin(θyz)
-      coszx = cos(θzx)
-      sinzx = sin(θzx)
 
-      _1 = one(T)
-      _0 = zero(T)
+    # ellipsoid matrix
+    Q = themes[convention].extrinsic ? P*Λ*P' : P'*Λ*P
 
-      Rxy = [cosxy -sinxy _0
-             sinxy  cosxy _0
-             _0     _0 _1]
+    new(Mahalanobis(Q))
+  end
 
-      Ryz = [_1    _0     _0
-             _0 cosyz -sinyz
-             _0 sinyz  cosyz]
+  function Ellipsoidal{N,T}(semiaxes, quaternion::Quaternion) where {N,T}
```

Comment:
I vote for just removing this constructor for now, unless you are seeing a good reason to keep it in practice. My intuition says that not many people would use it.

Reply:
I'm not used to work with quaternions either. Thought about putting it in case some user prefer working with them. But I agree with you. In case someone is working with quaternions, it's possible to easily convert it to angles via ReferenceFrameRotations.jl `quat_to_angles()`